src/Build/BuildCheck/Infrastructure/ConfigurationProvider.cs

```diff
@@ -140,6 +136,33 @@ public static void CheckCustomConfigurationDataValidity(string projectFullPath,
         return configurations;
     }
 
+    internal Dictionary<string, string> GetConfiguration(string projectFullPath, string ruleId)
+    {
+        var config = new Dictionary<string, string>();
```

Comment:
This is an unnecessary allocation.

```suggestion
        Dictionary<string, string> config;
```

Reply:
Updated, thank you